assets/src/edit-story/components/postLock/postLock.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { trackError } from '@web-stories-wp/tracking';
+
+/**
+ * Internal dependencies
+ */
+import { useAPI } from '../../app/api';
+import { useConfig } from '../../app/config';
+import PostLockDialog from './postLockDialog';
+
+function PostLock() {
+  const {
+    actions: { getStoryLockById, setStoryLockById },
+  } = useAPI();
+  const {
+    storyId,
+    userId,
+    previewLink,
```

Comment:
Get `previewLink` from `useStory` instead.


```js
const { previewLink } = useStory(
    ({
      state: {
        story: { previewLink },
      },
    }) => ({ previewLink })
  );
```

Reply:
This didn't work in my testing for some reason. I will loop around on this.

Reply:
to get this to work,  I had to move this compontent into inside the story provider. I have done this, but I am not sure this is the best idea...

Reply:
Changed here, https://github.com/google/web-stories-wp/pull/6497/commits/f173b16d90ef3547fc7f1ce50c8cbb8f75e80ab6 Not sure this is the best idea personally. But it works.